### PR TITLE
Passport

### PR DESCRIPTION
**Pull request by vp9920**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/179%23issuecomment-174769401%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/179%23discussion_r50819323%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/179%23discussion_r50820168%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/179%23discussion_r50820347%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/179%23issuecomment-174769401%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22That%27s%20it.%20If%20everything%20all%20right%2C%20I%27ll%20write%20unit%20tests%20and%20delete%20console.log%22%2C%20%22created_at%22%3A%20%222016-01-26T01%3A52%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11821757%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/vp9920%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20788cfaef94bcc174b4f26c521e7e267ec2e2a19c%20TCMSApp/src/server/model/mschemas/user.js%2028%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/179%23discussion_r50819323%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22it%20seems%20that%20you%27ve%20forgotten%20to%20merge%20with%20master.%20we%20already%20have%20this%20code%20there%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-01-26T10%3A46%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/server/model/mschemas/user.js%3AL1-33%22%7D%2C%20%22Pull%20788cfaef94bcc174b4f26c521e7e267ec2e2a19c%20TCMSApp/src/server/routes/login.js%2047%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/179%23discussion_r50820168%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22it%20seems%20that%20you%27re%20creating%20the%20user%20in%20DB%20here%20during%20trello%20login.%20But%20I%20am%20entering%20my%20local%20mongo%20db%20and%20don%27t%20see%20any%20user.%5Cr%5CnDo%20you%20know%20why%3F%20%22%2C%20%22created_at%22%3A%20%222016-01-26T10%3A55%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22in%20other%20words%2C%20what%20is%20%60if%20%28profile%29%60%20%3F%20%3A%29%20%22%2C%20%22created_at%22%3A%20%222016-01-26T10%3A57%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/server/routes/login.js%3AL1-100%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/179#issuecomment-174769401'>General Comment</a></b>
- <a href='https://github.com/vp9920'></a> That's it. If everything all right, I'll write unit tests and delete console.log
- [ ] <a href='#crh-comment-Pull 788cfaef94bcc174b4f26c521e7e267ec2e2a19c TCMSApp/src/server/model/mschemas/user.js 28'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/179#discussion_r50819323'>File: TCMSApp/src/server/model/mschemas/user.js:L1-33</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> it seems that you've forgotten to merge with master. we already have this code there :)
- [ ] <a href='#crh-comment-Pull 788cfaef94bcc174b4f26c521e7e267ec2e2a19c TCMSApp/src/server/routes/login.js 47'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/179#discussion_r50820168'>File: TCMSApp/src/server/routes/login.js:L1-100</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> it seems that you're creating the user in DB here during trello login. But I am entering my local mongo db and don't see any user.
Do you know why?
- <a href='https://github.com/ITsvetkoFF'></a> in other words, what is `if (profile)` ? :)

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-012/pull/179?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-012/pull/179?approve=1'></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-012/pull/179'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ITsvetkoFF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ITsvetkoFF/Kv-012/pull/179'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by vp9920** [comment]
That's it. If everything all right, I'll write unit tests and delete console.log

File: TCMSApp/src/server/model/mschemas/user.js [hunk 1]
```diff
@@ -1,48 +1,32 @@
...
+    currentProjectID: String,
+    trelloUserID: String,
     email: {
         type: String,
-        required: true,
         lowercase: true,
         validate: {
-            //TODO: make better regular expression for email. consider https://github.com/chriso/validator.js
             validator: function (v) {
-                return /^[-a-z0-9~!$%^&*_=+}{\'?]+(\.[-a-z0-9~!$%^&*_=+}{\'?]+)*@([a-z0-9_][-a-z0-9_]*(\.[-a-z0-9_]+)*\.(aero|arpa|biz|com|coop|edu|gov|info|int|mil|museum|name|net|org|pro|travel|mobi|[a-z][a-z])|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i.test(v); // jshint ignore:line
```
Comment (ITsvetkoFF) [inline TCMSApp/src/server/model/mschemas/user.js]: it seems that you've forgotten to merge with master. we already have this code there :)

File: TCMSApp/src/server/routes/login.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    if (user) {
+                        console.log('Everything is ok');
+                        return done(null, user);
+                    } else {
+                        var newUser = new User();
+                        newUser.fullName = profile.displayName;
+                        newUser.role = 'administrator';
+                        newUser.trelloUserID = profile.id;
+
+                        newUser.save(function(err) {
```
Comment (ITsvetkoFF) [inline TCMSApp/src/server/routes/login.js]: it seems that you're creating the user in DB here during trello login. But I am entering my local mongo db and don't see any user.
Do you know why?

Comment (ITsvetkoFF) [inline TCMSApp/src/server/routes/login.js]: in other words, what is `if (profile)` ? :)

**Pull request closed without merging**